util/topsql/reporter/reporter.go

```diff
@@ -485,6 +520,35 @@ func (tsr *RemoteTopSQLReporter) reportWorker() {
 	}
 }
 
+// removeDownDataSinks removes all down dataSinks
+func (tsr *RemoteTopSQLReporter) removeDownDataSinks() {
+	idx := 0
+	for _, dataSink := range tsr.dataSinks {
+		if dataSink.IsDown() {
+			dataSink.Close()
```

Comment:
Does this mean that it's dataSink's should only mark itself as down (but not clean up resources) and only clean up resource when called by Close?

Reply:
I rethought just before. `Close` is just a notification to `DataSink` that the Reporter is dropping it.